x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -606,14 +605,14 @@ public Builder(DatafeedConfig config) {
             this.jobId = config.jobId;
             this.queryDelay = config.queryDelay;
             this.frequency = config.frequency;
-            this.indices = config.indices;
-            this.types = config.types;
-            this.query = config.query;
-            this.aggregations = config.aggregations;
-            this.scriptFields = config.scriptFields;
+            this.indices = new ArrayList<>(config.indices);
+            this.types = new ArrayList<>(config.types);
+            this.query = config.query == null ? null : new HashMap<>(config.query);
+            this.aggregations = config.aggregations == null ? null : new HashMap<>(config.aggregations);
```

Comment:
I think `query` and `aggregations` should use `LinkedHashMap` to retain the original order.